plugins/plugin-chart-echarts/src/BoxPlot/transformProps.ts

```diff
@@ -0,0 +1,178 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  CategoricalColorNamespace,
+  ChartProps,
+  convertMetric,
+  DataRecord,
+  getNumberFormatter,
+  QueryFormDataMetric,
+} from '@superset-ui/core';
+import { BoxPlotQueryFormData } from './types';
+import { EchartsProps } from '../types';
+import { extractGroupbyLabel } from '../utils/series';
+
+export default function transformProps(chartProps: ChartProps): EchartsProps {
+  const { width, height, formData, queryData } = chartProps;
+  const data: DataRecord[] = queryData.data || [];
+  const {
+    colorScheme,
+    groupby = [],
+    metrics: formDataMetrics,
+    numberFormat,
+    xTicksLayout,
+  } = formData as BoxPlotQueryFormData;
+  const colorFn = CategoricalColorNamespace.getScale(colorScheme as string);
+  const numberFormatter = getNumberFormatter(numberFormat);
+
+  // TODO: remove cast once metrics is cleaned up in QueryFormData
```

Comment:
Just making sure this isn't something you'd intended to tackle in this PR ;)

Reply:
This requires a separate PR, as it will touch on many unit tests in `superset-ui/core`.